src/ahp/mod.rs

```diff
@@ -59,6 +59,19 @@ impl<F: PrimeField> AHPForR1CS<F> {
         .max()
         .unwrap())
     }
+
+    // Get the degree bounds needed to trim the prover and verifier keys for the index.
+    pub fn get_degree_bounds(info: &indexer::IndexInfo<F, C>) -> [usize; 3] {
+        let degree_bounds: [usize; 3];
+        let num_constraints = info.num_constraints;
+        let num_non_zero = info.num_non_zero;
+        let h_size = EvaluationDomain::<F>::compute_size_of_domain(num_constraints).unwrap();
+        let k_size = EvaluationDomain::<F>::compute_size_of_domain(num_non_zero).unwrap();
+
+        degree_bounds[0] = h_size - 2;
+        degree_bounds[1] = h_size - 2;
+        degree_bounds[2] = k_size - 2;
+        degree_bounds
```

Comment:
Maybe change this to something like this:
```rust
let degree_bounds = vec![None; index_vk.index_comms.len()]
            .into_iter()
            .chain(AHPForR1CS::prover_first_round_degree_bounds(&index_vk.index_info))
            .chain(AHPForR1CS::prover_second_round_degree_bounds(&index_vk.index_info))
            .chain(AHPForR1CS::prover_third_round_degree_bounds(&index_vk.index_info))
            .chain(AHPForR1CS::prover_fourth_round_degree_bounds(&index_vk.index_info))
			.filter(|d_b| d_b.is_some())
			.map(|d_b| d_b.unwrap())
            .collect::<Vec<_>>();
```

This would unify it with the logic [here](https://github.com/scipr-lab/marlin/blob/2f64b407c9ee00f4666ad1996465245d4c5bc899/src/lib.rs#L307)

(Even better would be to make a common function that's called in both places, but we can do that in a follow-up PR)

Reply:
Don't want to change this for now. I think this should be done as a larger refactor, where I agree a common function should be called.